aten/src/ATen/cpu/vec256/vec256_base.h

```diff
@@ -64,115 +66,199 @@ struct Vec256 {
     std::memcpy(vec.values, ptr, count * sizeof(T));
     return vec;
   }
-  void store(void* ptr, int count = size) const {
+  void storeu(void* ptr, int count = size) const {
     std::memcpy(ptr, values, count * sizeof(T));
   }
-  Vec256<T> map(T (*f)(T)) const {
-    Vec256<T> ret;
-    for (int64_t i = 0; i != size; i++) {
-      ret.values[i] = f(values[i]);
-    }
-    return ret;
-  }
-  Vec256<T> abs() const {
-    Vec256<T> ret;
-    for (int64_t i = 0; i < size; i++) {
-      ret.values[i] = values[i] < 0 ? -values[i] : values[i];
-    }
-    return ret;
-  }
-  Vec256<T> acos() const {
-    return map(std::acos);
-  }
-  Vec256<T> asin() const {
-    return map(std::asin);
-  }
-  Vec256<T> atan() const {
-    return map(std::atan);
-  }
-  Vec256<T> erf() const {
-    return map(std::erf);
-  }
-  Vec256<T> exp() const {
-    return map(std::exp);
-  }
-  Vec256<T> expm1() const {
-    return map(std::expm1);
-  }
-  Vec256<T> log() const {
-    return map(std::log);
-  }
-  Vec256<T> log10() const {
-    return map(std::log10);
+  const T& operator [](int idx) const {
+    return values[idx];
   }
-  Vec256<T> log1p() const {
-    return map(std::log1p);
+  void set_value(int64_t idx, T value) {
+    values[idx] = value;
   }
-  Vec256<T> log2() const {
-    return map(std::log2);
-  }
-  Vec256<T> ceil() const {
-    return map(std::ceil);
-  }
-  Vec256<T> cos() const {
-    return map(std::cos);
-  }
-  Vec256<T> floor() const {
-    return map(std::floor);
-  }
-  Vec256<T> round() const {
-    return map(std::round);
-  }
-  Vec256<T> sin() const {
-    return map(std::sin);
+};
+template <class T>Vec256<T> map(T (*f)(T), Vec256<T> x) {
+  Vec256<T> ret;
+  for (int64_t i = 0; i != Vec256<T>::size; i++) {
+    ret.set_value(i, f(x[i]));
   }
-  Vec256<T> tanh() const {
-    return map(std::tanh);
+  return ret;
+}
+
+template <class T>Vec256<T> abs(Vec256<T> x) {
+  Vec256<T> ret;
+  for (int64_t i = 0; i < Vec256<T>::size; i++) {
+    ret.set_value(i, x[i] < 0 ? -x[i] : x[i]);
   }
-  Vec256<T> trunc() const {
-    return map(std::trunc);
+  return ret;
+}
+
+template <class T>
+Vec256<T> acos(Vec256<T> x) {
```

Comment:
Why did we suddenly stopped using those as methods? This really unnecessarily increases those diffs and makes it so much harder to review them...

Reply:
I'll revert